hyperspy/_signals/hologram_image.py

```diff
@@ -0,0 +1,303 @@
+            # if hasattr(f.canvas.manager, 'window'):
+            #     f.canvas.manager.window.raise_()
+            # plt.waitforbuttonpress(100)
+            # plt.waitforbuttonpress(5)
+            # # Use this one in the case of usage of rect_roi obj:
+            # arect = eh_hw_fft[rect.y0:rect.y1, rect.x0:rect.x1]
+            # # Sideband position,find the max number and its [c,r]
+            # yR, xC = np.unravel_index(arect.argmax(), arect.shape)  # Center of selected sideBand
+            # # Use this one in the case of usage of rect_roi obj
+            # sb_pos = [np.round(rect.y0) + yR, np.round(rect.x0) + xC]
+            # a = 1.0
+            # sb_size = np.round(np.array([a / 3, a / 2, a]) *
+            #                    (norm(np.subtract(sb_pos, [sx / 2, sy / 2])) * np.sqrt(2)))
+            # # To be sure of even number, still questionable if it is needed:
+            # sb_size = sb_size - np.mod(sb_size, 2)
+            # print("Sideband range in pixels")
+            # print("%d %d %d" % (sb_size[0], sb_size[1], sb_size[2]))
+            # # Choose SideBand size
+            # sb_size = input("Choose Sideband size  pixel = ")
+            # sb_size = sb_size - np.mod(sb_size, 2)  # to be sure of even number...
+            # print("Sideband Size in pixels")
+            # print("%d" % sb_size)
+            # plt.close(f)
+            # rec_param = (rect.x0, rect.y0, rect.x1, rect.y1, sb_size)
+        else:
+            arect = eh_hw_fft[rec_param[1]:rec_param[3], rec_param[0]:rec_param[2]]
+            # Sideband position,find the max number and its [c,r]
+            yr, xc = np.unravel_index(arect.argmax(),
+                                      arect.shape)  # center of the selected sideBand
+            sb_pos = [np.round(rec_param[1]) + yr, np.round(
+                rec_param[0]) + xc]  # <----- use this one in the case of usage of rect_roi obj
+            sb_size = rec_param[4]
+
+        if verbose:
+            plt.figure()
+            plt.imshow(np.log(np.abs(eh_hw_fft)))
+            plt.hold(True)
+            plt.plot(sb_pos[1], sb_pos[0], 'r+')
+
+        # Reconstruction
+        if fresnel_fringe_filter:
+            fresnel_width = None
+        else:
+            fresnel_width = 0
+
+        if ref_data is None:
+            w_ref = 1
+        else:
+            w_ref = self._reconstruct(ref_data.data, sb_size, sb_pos, fresnel_width=fresnel_width)  # reference electron wave
+
+        w_obj = self._reconstruct(holo_data, sb_size, sb_pos, fresnel_width=fresnel_width)  # object wave
+
+        wave = w_obj / w_ref
+        wave_image = self._deepcopy_with_new_data(wave)
+        wave_image.set_signal_type('wave')  # New signal is normal image without special signal type!
+        rec_param_dict = {'sb_pos_x0': rec_param[0], 'sb_pos_x1': rec_param[1],
+                          'sb_pos_y0': rec_param[2], 'sb_pos_y1': rec_param[3],
+                          'sb_size': rec_param[4]}
```

Comment:
OrderedDict does it job, but it seems that it gets screwed up after retrieving it from metadata, so I've sorted the values manual in the get_rec_param  function for now.